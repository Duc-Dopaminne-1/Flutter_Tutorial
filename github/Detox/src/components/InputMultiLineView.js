import React from 'react';
import {StyleSheet, TextInput, View} from 'react-native';

const styles = StyleSheet.create({
  multilineInput: {
    textAlignVertical: 'top',
  },
});

const InputMultiLineView = ({
  placeholderTextColor,
  placeholder,
  onChangeText,
  keyboardType,
  value,
  editable,
  customStyle,
  maxLength,
  multiline = true,
  ...textInputProps
}) => {
  return (
    <View>
      <TextInput
        placeholderTextColor={placeholderTextColor}
        placeholder={placeholder}
        onChangeText={onChangeText}
        keyboardType={keyboardType}
        value={value}
        editable={editable}
        autoCorrect={false}
        autoCapitalize="none"
        maxLength={maxLength}
        multiline={multiline}
        {...textInputProps}
        style={[textInputProps.style, styles.multilineInput]}
      />
    </View>
  );
};

export default InputMultiLineView;
