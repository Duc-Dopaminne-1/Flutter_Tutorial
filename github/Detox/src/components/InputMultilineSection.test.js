import React from 'react';
import renderer from 'react-test-renderer';

import InputMultiLineView from './InputMultiLineView';

it('renders correctly', () => {
  const tree = renderer.create(<InputMultiLineView />).toJSON();
  expect(tree).toMatchSnapshot();
});
