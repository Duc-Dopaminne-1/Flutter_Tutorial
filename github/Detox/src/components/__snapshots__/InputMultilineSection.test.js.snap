// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    multiline={true}
    style={
      Array [
        undefined,
        Object {
          "textAlignVertical": "top",
        },
      ]
    }
  />
</View>
`;
